build.cmd

```diff
@@ -8,9 +8,46 @@ setlocal
 
 :ReadArguments
 :: Read in the args to determine whether to run the native build, managed build, or both (default)
-set "__args=%*"
-if /i [%1] == [native] (set __buildSpec=native&&set "__args=%__args:~6%"&&shift&&goto Tools)
-if /i [%1] == [managed] (set __buildSpec=managed&&set "__args=%__args:~7%"&&shift&&goto Tools)
+set OfficialBuildIdArg=
+set "__args= %*"
+set processedArgs=
+set unprocessedBuildArgs=
+
+:Loop
+if [%1]==[] goto Tools
+
+if /I [%1]==[native] (
+    set __buildSpec=native
+    set processedArgs=!processedArgs! %1
+    goto Next
+)
+
+if /I [%1] == [managed] (
+    set __buildSpec=managed
+    set processedArgs=!processedArgs! %1
+    goto Next
+)
+
+if /I [%1] == [officialbuildid] (
```

Comment:
Can we still make this `[/p:OfficalBuildId]`? I would like to make the way we pass this consistent.